### ISSUE TITLE
Config option to enable/disable batchnorm running values

### ISSUE THREAD

**Issue by josedvq** [speediedan/finetuning-scheduler#13]
Thanks for sharing this great tool

## 🚀 Feature

Add constructor argument `batch_norm_track_running_stats` to control `track_running_stats` of frozen batch norm layers, or avoid changing `track_running_stats` of batchnorm layers.

### Motivation

Currently FinetuningSchedular calls `self.freeze(modules=pl_module, train_bn=False)`. That in turns executes this code for each model module:

```
if isinstance(module, _BatchNorm):
    module.track_running_stats = False
# recursion could yield duplicate parameters for parent modules w/ parameters so disabling it
for param in module.parameters(recurse=False):
    param.requires_grad = False
```
This can be confusing because setting `track_running_stats=False` is not normally done when fine-tunning. It also silently changes the values set when creating the model. The default value of `track_running_stats` is True so when one fine-tunes a model following the standard procedure of setting `requires_grad` normally `track_running_stats` will be True unless set otherwise. I'm not sure why it is implemented this way in `BaseFinetuning`

### Pitch

Adding a constructor argument to provide control over the batchnorm behavior when using the module. I think the default should be to set `track_running_stats=True` to reproduce the standard "recipe" for fine-tuning.
Alternatively do not alter track_running_stats at all.
At the very least I think it would be important to document the current behavior. 

______________________________________________________________________

**Comment by speediedan** [comment]
Thanks for your valuable suggestion/feedback @josedvq! Apologies for the delayed response, I've have limited bandwidth to dedicate to Finetuning Scheduler between minor releases so am just preparing FTS for the upcoming 2.3.0 Lightning release now.

### Root Cause

About 19 months ago FTS set a default of `train_bn=False` during layer freezing to address issue [#5](https://github.com/speediedan/finetuning-scheduler/issues/5) handing control of `BatchNorm` freezing to the FTS schedule while leaving `track_running_stats` `True` for the frozen BatchNorm layers by default. About a month subsequently, `BaseFinetuning` changed this default behavior and began to set [`track_running_stats` to `False` upon freezing BatchNorm layers](https://github.com/Lightning-AI/pytorch-lightning/pull/15063). Given that there are use cases for both settings `track_running_stats` settings, I think a constructor parameter like the one you suggest is a great idea. 

In light of your suggestion I've just [pushed a commit](9d7014bde5b7963f10dd2b35714c25af59794f7c) that I think accordingly improves FTS usability...

#### New State w/ Usability Improvements - (FTS >= v2.3.0) 

[The commit](9d7014bde5b7963f10dd2b35714c25af59794f7c) will be released with the next Lightning/FTS minor release (``2.3.0``). This commit introduces the [`frozen_bn_track_running_stats` option](https://finetuning-scheduler.readthedocs.io/en/latest/api/finetuning_scheduler.fts.html#finetuning_scheduler.fts.FinetuningScheduler.params.frozen_bn_track_running_stats) to the FTS callback constructor, allowing the user to override the default Lightning behavior that disables `track_running_stats` when freezing `BatchNorm` layers. To avoid changing the current behavior without warning, I've made the default for this parameter ``False`` but am warning users the default value will be ``True`` [starting with FTS release](https://finetuning-scheduler.readthedocs.io/en/latest/#:~:text=Beginning%20with%20FTS%202.4.0) ``2.4.0``. 

Hope you've found FTS of some utility once working through the issue you encountered. 

Thanks again for your contribution, you've helped improve FTS for everyone! Feel free to reach out anytime if you have other issues or want to share more about your use case. Best of luck with your work!

**Comment by speediedan** [comment]
fixed with commit 9d7014bde5b7963f10dd2b35714c25af59794f7c
